### DISCUSSION TITLE
Will overlay / overlay-tmpfs options ever be restored (overlayfs)?

### DISCUSSION THREAD

**Discussion by cyberroneous** [netblue30/firejail#6041] [Q&A] [answered]
I understand these two options (--overlay and --overlay-tmpfs) were removed a while back due to a bug which allowed for root privilege escalation. I also believe though (?) that the bug has since been fixed, yet these overlay options remain disabled in the configure.ac file of the current build:

```
# overlayfs features temporarily disabled pending fixes
HAVE_OVERLAYFS=""
AC_SUBST([HAVE_OVERLAYFS])
#AC_ARG_ENABLE([overlayfs],
#    [AS_HELP_STRING([--disable-overlayfs], [disable overlayfs])])
#AS_IF([test "x$enable_overlayfs" != "xno"], [
#	HAVE_OVERLAYFS="-DHAVE_OVERLAYFS"
#])

``` 

Are these options scheduled to be restored/reverted at some point, or has the corresponding bug not yet been fixed?

Relates to:

* #4178
* #6632

**Comment by rusty-snake** [comment, marked as answer]
See security bug (and all the other bugs with it) are not fixed yet. See #4178 for futher discussions in overlayfs.

**Reply by rusty-snake** [reply]
> scheduled to be restored/reverted at some point

Scheduled to be restored when some foss contributor/developer takes the time and knowledge and fixes the overlayfs code :smile:.
